src/transformers/pipelines/base.py

```diff
@@ -277,7 +277,8 @@ def infer_framework_load_model(
         if isinstance(model, str):
             raise ValueError(f"Could not load model {model} with any of the following classes: {class_tuple}.")
 
-    framework = infer_framework(model.__class__)
+    if framework is None:
+        framework = infer_framework(model.__class__)
```

Comment:
The fix is nice, however, we shouldn't even call this function if both model and framework are specified (since we have all the infiormation to start running the pipeline already)

Reply:
Do you mind checking/fixing why this function is called in the first place if `not isinstance(model, str)` (means you send a class bascially, we should just use it as-is) ? If we can fix there it would be even cleaner imo.